### PR TITLE
Android fullscreen via bridging

### PR DESCRIPTION
**Pull request by ShariqMush**
Native bridge to android launching android video view in fullscreen, hours of work of sweat and tears!!!
To all the other developers, please use [this](https://github.com/ShariqMush/react-native-fullscreenvideo-player) github library for working fullscreen until the developer of this library merges the pull request :) I will be updating that fork as well.

### CODE REVIEW

**Comment by cornedor** [comment]
Thanks for this commit.
Do you by any change have the rest of the Android project?

**Comment by ShariqMush** [comment]
@cornedor Nope, readme says it all.

**Comment by cornedor** [comment]
Ah I didn't see that, thanks. I'll try to make this work with react-native link.

**Comment by ShariqMush** [comment]
@cornedor Can you check my pull request? It is native and performs better than react because in react you would have to resize the video component which is unstable where native video player fixes it all in native so please check my request and merge it. Thank you.

**Comment by cornedor** [comment]
@ShariqMush yes I agree, I have more time after next week, so I will try to make it work with react-native link. If you update the app now you will need to manually copy the files from the repo to your own project.

I do appreciate all the work you did to get it this far.

**Comment by abrown1982** [comment]
I spent the last few days looking for the best full-screen android option for RN and I think this pull request is the best available.  A few issues though.

1 - No option to leave fullscreen (other than the back button) once in fullscreen.
2 - Position of video is not retain when switching between inline and fullscreen.

I have written fixes to both of these issues (doubleclick on fullscreen to close and passing of position between activities), though I have not tested it much, would appreciate someone else running their eyes over it to make sure I've not missed anything?

https://github.com/abrown1982/react-native-video-player/commit/43bffe53e747a13a1db712c40ab917a296c791ee

**Comment by Misiur** [comment]
This works great, but fullscreen has native controls, not the react-native-video-player controls. Can something be done about this, or is it too compilcated?

**Comment by ShariqMush** [comment]
@Misiur Yes it is very complicated, why would you need the react controls when you are already getting the native controls?

**Comment by Misiur** [comment]
Missing bacl-from-fullscreen button, missing volume controls, just all around missing functionality - that's why I'm using react-native-video-player in the first place! (well, and your awesome native fulllscreen functionality)

**Comment by ShariqMush** [comment]
@Misiur For the back button, why not use the ordinary back button on any android device lol but I can look at the volume controls for you but then its pretty much like the player rebuilt into the native but yeah I will look forward to it but please post an issue on my [fork](https://github.com/ShariqMush/react-native-fullscreenvideo-player). I will be posting further updates there until this pull request gets accepted.

**Comment by rnowm** [comment]
Any update on that one? would be great if you could merge it, I was not able to make it work following the described steps. Thanks!

File: android/app/src/main/java/BridgePackage.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.uimanager.ViewManager;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class BridgePackage implements ReactPackage {
+
+    @Override
```
Comment (linesh-simplicity) [inline android/app/src/main/java/BridgePackage.java]: [from RN 0.47.x on](https://github.com/facebook/react-native/releases/tag/v0.47.2) the `createJSModules()` method has been removed and thus the `@Override` implement here will fail the gradle compilation. [Simply remove the `@Override` line and keep the method call](https://github.com/facebook/react-native/commit/ce6fb337a146e6f261f2afb564aa19363774a7a8) should solve this issue.

File: index.js [hunk 2]
```diff
@@ -1,9 +1,7 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import { Image, Platform, StyleSheet, TouchableOpacity, View, ViewPropTypes } from 'react-native';
+import React, { Component, PropTypes } from 'react';
+import { View, StyleSheet, Image, TouchableOpacity, Platform, NativeModules } from 'react-native';
```
Comment (linesh-simplicity) [inline index.js]: Any specific reasons for changing to use the built-in `PropTypes` instead of the separate `prop-types` library?

**Comment by JayricMok** [comment]
@Misiur I agree with you, native control don't have back fullscreen, and the control are hard to make a custom control bar like without fullscreen

**Comment by solonik1** [comment]
+1

**Comment by Angelk90** [comment]
@ShariqMush , @abrown1982: Ot, I have a problem, I created a native module, but when you start the native module all right, it works. But if the user presses the physical back button, the app crashes, how can I fix this? I would like when the user presses the physical back button, he returns to the native react app at the point I had left before the native module started.

File: index.js (hunk 2 above)
Comment (kashifaliquazi) [reply index.js]: built-in PropTypes are breaking since react does not support them, we should use separate prop-types library.

**Comment by kashifaliquazi** [comment]
Native player is working fine but how could I get the callbacks like when user exits from fullscreen "onFullscreenPlayerDidDismiss" in android

**Comment by kashifaliquazi** [comment]
events like onFullscreenPlayerDidDismiss are not working in android

**Comment by kashifaliquazi** [comment]
@abrown1982 can you pleae help me in adding callbacks(onEnd,onFullscreenPlayerDidDismiss etc ) on full screen mode. 
thanks in Advance

**Comment by abrown1982** [comment]
@kashifaliquazi I'm working to a deadline at the moment so don't have any time available in the next week or so.

File: android/app/src/main/java/BridgePackage.java (hunk 1 above)
Comment (ShariqMush) [reply android/app/src/main/java/BridgePackage.java]: Thanks man!

**Comment by ShariqMush** [comment]
I have been away from my computer but I'll be updating the fork asap with the callbacks.

**Comment by Ilario17** [comment]
@ShariqMush I installed your library but the fullscreen button is not present anyway. 
@cornedor Would not it be better to keep a single library?

**Comment by hoscarcito** [comment]
It's been a month since the last update here... Any news?

**Comment by SujaYuga** [comment]
I too tried with PR#38 no errors but the fullscreen button on Android does not show. I am using react native video player version #0.9.1. Any update please?

**Comment by xstable** [comment]
Would love to see the FullScreen work.

**Comment by ShariqMush** [comment]
@hoscarcito @SujaYuga @xstable @Ilario17 I have been inactive for a while but will look at what the problem is very soon. Check forks of my own fork and there you should find some working ones.

**Comment by xstable** [comment]
@ShariqMush Any progress here?

**Comment by ShariqMush** [comment]
@xstable Hi, I am not working with react native at the moment but I believe that the issue is related to react native library not any problem in the native code. If fullscreen button does not appear then you should tamper with the library files. Try pulling my fork of the video player to see if it works because some people made some PR's that I merged recently.

**Comment by xstable** [comment]
> I spent the last few days looking for the best full-screen android option for RN and I think this pull request is the best available. A few issues though.
> 
> 1 - No option to leave fullscreen (other than the back button) once in fullscreen.
> 2 - Position of video is not retain when switching between inline and fullscreen.
> 
> I have written fixes to both of these issues (doubleclick on fullscreen to close and passing of position between activities), though I have not tested it much, would appreciate someone else running their eyes over it to make sure I've not missed anything?
> 
> [abrown1982@43bffe5](https://github.com/abrown1982/react-native-video-player/commit/43bffe53e747a13a1db712c40ab917a296c791ee)

@abrown1982 is there a chance to have a classic  "x" as close-button in the Fullscreen-View?
That would be awesome.

**Comment by muhammadwfa** [comment]
is there any update for full screen mode

**Comment by vlatri** [comment]
Hey guys! Any news? We're getting tired of stretching video components manually.

**Comment by iamshadmirza** [comment]
> 
> 
> Hey guys! Any news? We're getting tired of stretching video components manually.

We need fullscreen feature on Android.

**Comment by ShariqMush** [comment]
@iamshadmirza Isn't the issue with the fullscreen fixed yet? I am not working with react native anymore :)

**Comment by nickarora** [comment]
If this was approved, why hasn't this been merged in yet? @cornedor is this repo still maintained?

**Comment by ayeshasaeed94** [comment]
@ShariqMush any updates regarding full screen mode on android?

**Comment by RishavKumar-3796** [comment]
Please fix this ASAP

**Comment by TheRav3n** [comment]
I verified that this PR is not working out of the box, not even after following the complex guidelines added in the README.md file. Will be looking into this and keep you all updated.

**Comment by 15110011** [comment]
tried to config like [this](https://github.com/ShariqMush/react-native-fullscreenvideo-player#setting-up-fullscreen-on-android) and still not working

**Comment by pierroo** [comment]
It's a bit old but: is there any news regarding fullscreen for android?

**Comment by moskalakamil** [comment]
Hey, right now we upgraded react-native-version in this library and fullscreen is added there. I will close it, but i would like to thank you so much for making this!

**Pull request closed without merging**